### ISSUE TITLE
Add support of `update_or_create`

### ISSUE THREAD

**Issue by dorosch** [fedora-infra/sqlalchemy-helpers#422]
Thanks for your project, it's absolutely amazing 

Could you please consider adding a function `update_or_create`?

Thanks

**Comment by abompard** [comment]
It seems fine, yeah, I suppose you want to mimic [Django's `update_or_create`](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#update-or-create)?

**Comment by dorosch** [comment]
Yes exactly. You already have [a similar function](https://github.com/fedora-infra/sqlalchemy-helpers/blob/develop/sqlalchemy_helpers/aio.py#L199)
